config/v1/types_authentication.go

```diff
@@ -771,3 +826,46 @@ type TokenRequiredClaim struct {
 	// +required
 	RequiredValue string `json:"requiredValue"`
 }
+
+type TokenClaimValidationCELRule struct {
+	// expression is a CEL expression evaluated against token claims.
+	// The expression must be a non-empty string and no longer than 1024 characters.
+	// The expression must return a boolean value where 'true' signals a valid token and 'false' an invalid one.
+	// This field is required.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=1024
+	// +required
+	Expression string `json:"expression,omitempty"`
+
+	// message allows configuring a human-readable message that is logged by the Kubernetes API server
+	// when a token fails validation based on the CEL expression defined in 'Expression'.
+	// This field is required. If provided, the message must be at least 1 character long
+	// and cannot exceed 256 characters. This message is logged and not returned to the caller.
```

Comment:
"This field is required. If provided ..." reads weird. If the field is required it must be provided.

Maybe:
```suggestion
	// message is required, must be at least 1 character long and must not exceed 256 characters.
```
?